app/scripts/transaction-manager.js

```diff
@@ -137,25 +137,30 @@ module.exports = class TransactionManager extends EventEmitter {
   }
 
   // formats txParams so the keyringController can sign it
-  formatTxForSigining (txParams, cb) {
-    var address = txParams.from
-    var metaTx = this.getTx(txParams.metamaskId)
-    var gasMultiplier = metaTx.gasMultiplier
-    var gasPrice = new BN(ethUtil.stripHexPrefix(txParams.gasPrice), 16)
-    gasPrice = gasPrice.mul(new BN(gasMultiplier * 100, 10)).div(new BN(100, 10))
-    txParams.gasPrice = ethUtil.intToHex(gasPrice.toNumber())
-
-    // normalize values
-    txParams.to = normalize(txParams.to)
-    txParams.from = normalize(txParams.from)
-    txParams.value = normalize(txParams.value)
-    txParams.data = normalize(txParams.data)
-    txParams.gasLimit = normalize(txParams.gasLimit || txParams.gas)
-    txParams.nonce = normalize(txParams.nonce)
-    const ethTx = new Transaction(txParams)
-
-    // listener is assigned in metamaskController
-    this.emit(`${txParams.metamaskId}:formatted`, ethTx, address, txParams.metamaskId, cb)
+  formatTxForSigining (txParams) {
+    return new Promise((resolve, reject) => {
+      try {
+        var address = txParams.from
+        var metaTx = this.getTx(txParams.metamaskId)
+        var gasMultiplier = metaTx.gasMultiplier
+        var gasPrice = new BN(ethUtil.stripHexPrefix(txParams.gasPrice), 16)
+        gasPrice = gasPrice.mul(new BN(gasMultiplier * 100, 10)).div(new BN(100, 10))
+        txParams.gasPrice = ethUtil.intToHex(gasPrice.toNumber())
+
+        // normalize values
+        txParams.to = normalize(txParams.to)
+        txParams.from = normalize(txParams.from)
+        txParams.value = normalize(txParams.value)
+        txParams.data = normalize(txParams.data)
+        txParams.gasLimit = normalize(txParams.gasLimit || txParams.gas)
+        txParams.nonce = normalize(txParams.nonce)
+        const ethTx = new Transaction(txParams)
+        var txId = txParams.metamaskId
+        resolve({ethTx, address, txId})
```

Comment:
An even simpler way to convert a flat function to promisified, is to simply `return Promise.resolve(yourValue)`.